openfisca_core/scenarios.py

```diff
@@ -230,10 +222,7 @@ def fill_simulation(self, simulation, use_set_input_hooks = True, variables_name
                                 array = holder.default_array()
                             array[axis['index']:: axis_entity.step_size] = axis['min'] \
                                 + mesh.reshape(steps_count) * (axis['max'] - axis['min']) / (axis_count - 1)
-                            if use_set_input_hooks:
-                                holder.set_input(axis_period, array)
-                            else:
-                                holder.put_in_cache(array, axis_period)
+                            holder.set_input(axis_period, array, behavior=SET_INPUT_ADD)
```

Comment:
Introducing this extra `behavior` parameter to resolve this buggy usecase is not great, but I understand we didn't find much better.

Let's keep moving forward, but I may be worth rethinking this before merging.